### PR TITLE
JAMES-3151 HasMimeType matchers fails on empty charset

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java [hunk 1]
```diff
@@ -124,4 +129,24 @@ public void matchShouldReturnRecipientsWhenAtLeastOneMimeTypeMatch() throws Exce
...
+
+    @Test
+    void matchShouldNotFailOnEmptyCharset() throws Exception {
+        matcher.init(FakeMatcherConfig.builder()
+            .matcherName("HasMimeType")
+            .condition("text/html")
+            .build());
+
+        MimeMessage message = mock(MimeMessage.class);
+        when(message.getHeader(CONTENT_TYPE)).thenReturn(new String[] {"text/html; charset="});
```
Comment (mbaechler) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java:141]: is it a legal value?

Comment (chibenwa) [reply mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java:141]: It is seen on the PROD... I bet we should not fail as technically the mimetype is specified.

Comment (chibenwa) [reply mailet/standard/src/test/java/org/apache/james/transport/matchers/HasMimeTypeTest.java:141]: Recovering from an illegal charset value seems legitimate to me.

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java [hunk 2]
```diff
@@ -58,19 +57,20 @@ public void init() throws javax.mail.MessagingException {
...
     public Collection<MailAddress> match(Mail mail) throws javax.mail.MessagingException {
-        Optional<String> mimeTypes = getMimeTypeFromMessage(mail.getMessage());
-
-        return mimeTypes.filter(acceptedContentTypes::contains)
-                .map(any -> mail.getRecipients())
-                .orElse(ImmutableList.of());
+        return StreamUtils.ofNullable(mail.getMessage().getHeader(CONTENT_TYPE))
+            .flatMap(this::getMimeType)
+            .filter(acceptedContentTypes::contains)
+            .findAny()
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java:63]: why are you using streams while you just need an Optional?

Comment (chibenwa) [reply mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java:63]: because there might be several content type header

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**